### ISSUE TITLE
[BUG] v2 BaseModel stores logging_metrics as plain list instead of nn.ModuleList (breaks GPU training)

### ISSUE THREAD

**Issue by StrikerEureka34** [sktime/pytorch-forecasting#2197]
Describe the bug

The v2 BaseModel stores logging_metrics as a plain Python list instead of wrapping it in nn.ModuleList. This means PyTorch doesn't register the metrics as submodules, so they don't move to GPU when the model does. On GPU training, this causes a device mismatch crash when `log_metrics()` runs.

v1 does this correctly in [_base_model.py](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/models/base/_base_model.py#L554) in nn.ModuleList. But v2 at [_base_model_v2.py](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/models/base/_base_model_v2.py#L56)  just assigns the raw list:

```python
# v2 (broken)
self.logging_metrics = logging_metrics if logging_metrics is not None else []

# v1 (correct)
self.logging_metrics = nn.ModuleList([...])
```

The metrics themselves (SMAPE, MAE, MAPE, etc.) all inherit from MultiHorizonMetric, which registers internal state buffers via [add_state()](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/metrics/base_metrics/_base_metrics.py#L829-L838) . These buffers stay on CPU when the model moves to GPU, because the metrics aren't registered submodules.

During a training/validation step, `log_metrics()` calls `metric(y_hat, y)`, which triggers `update()` -> `_update_losses_and_lengths()`, hitting:

```python
self.losses = self.losses + losses  # CPU buffer + GPU tensor -> RuntimeError
```

This affects all 5 existing v2 models (DLinear, TiDE, TimeXer, SAMformer, TFT) since they all inherit from BaseModel.

&ensp;
To Reproduce

```python
import torch
from pytorch_forecasting.metrics import MAE, SMAPE

from pytorch_forecasting.models.dlinear import DLinear

# Create a model with logging_metrics on GPU
model = DLinear(
    loss=MAE(),
    logging_metrics=[SMAPE(), MAE()],
    metadata={...},  # appropriate metadata
)
model = model.to("cuda")

# Check: logging_metrics did NOT move to GPU
for m in model.logging_metrics:
    for name, buf in m._buffers.items():
        print(f"{name}: {buf.device}")  # prints "cpu" -- should be "cuda"

# This will crash during training when log_metrics() is called
# because y_hat (cuda) and metric state buffers (cpu) are on different devices
```

The existing tests don't catch this because:

- The v2 integration suite (test_all_estimators_v2.py) only tests `fit()` and `predict()` at the pkg level, never calling `training_step()` directly with logging_metrics active
- Unit tests that pass ogging_metrics ([test_dlinear_v2.py](https://github.com/sktime/pytorch-forecasting/blob/main/tests/test_models/test_dlinear_v2.py#L118), [test_timexer_v2.py](https://github.com/sktime/pytorch-forecasting/blob/main/tests/test_models/test_timexer_v2.py#L208) only run `forward()` under `torch.no_grad()`, never `training_step()`
- CI runs on CPU, so the device mismatch never triggers

Expected behavior

logging_metrics should be stored as nn.ModuleList so that metrics are properly registered as submodules, moved to the correct device with the model, and included in `state_dict()` for checkpointing. This is how v1 handles it.

&ensp;
Suggested fix

In [_base_model_v2.py](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/models/base/_base_model_v2.py#L56):

```python
# before
self.logging_metrics = logging_metrics if logging_metrics is not None else []

# after
self.logging_metrics = nn.ModuleList(logging_metrics) if logging_metrics else nn.ModuleList()
```

And add a test that actually exercises `log_metrics()` during a training step with `ogging_metrics`passed, to prevent regression.

&ensp;

Additional context

Came across this while studying the v2 base layer for foundation model integration work #1959. 
FMs will train on GPU and need metric logging during fine-tuning, so this is directly in the path. 
Related to #1700 (logging redesign for v2) and #1754 (v1 device propagation for loss metrics), though neither covers this specific issue.

**Comment by fnhirwa** [comment]
hey @StrikerEureka34 this seems to be interesting find, however I can't be able to reproduce it from my end.

Would you provide a full reproducible example?

**Comment by StrikerEureka34** [comment]
> Would you provide a full reproducible example?


**Reproduced and validated on GPU.** 
I confirmed both states by stashing/restoring the fix.
Actually the original reproduction script I provided had metadata={...} as a placeholder so it couldn't be run directly. 
So I've added this script below which is fully self-contained and has no placeholders.

&ensp;

**My Environment for test**
```
Python:  3.10.12
PyTorch: 2.9.1+cu128
CUDA:    12.8
GPU:     NVIDIA A100 80GB PCIe
```

**Reproduction script** (no placeholder metadata, fully runnable):
```python
import numpy as np, pandas as pd, torch
from pytorch_forecasting.data.timeseries._timeseries_v2 import TimeSeries
from pytorch_forecasting.data._tslib_data_module import TslibDataModule
from pytorch_forecasting.metrics import MAE, SMAPE
from pytorch_forecasting.models.dlinear import DLinear

if not torch.cuda.is_available():
    raise SystemExit("CUDA not available, it must be run on a GPU machine for Verification")

n=64; t=np.arange(n)
df = pd.DataFrame({"time_idx": t, "series_id": 0, "value": np.sin(2 * np.pi * t / 20)})
ts = TimeSeries(df, time="time_idx", group=["series_id"], target=["value"],
                num=[], cat=[], known=["time_idx"], unknown=["value"])
dm = TslibDataModule(ts, context_length=16, prediction_length=4, batch_size=4)
dm.setup()
print("metadata:", dm.metadata)

model = DLinear(loss=MAE(), logging_metrics=[SMAPE(), MAE()], metadata=dm.metadata)
model = model.to("cuda")
print("type:", type(model.logging_metrics))
print("devices:", [m.losses.device for m in model.logging_metrics])
```

&ensp;

**Before fix** (upstream main(most recent)):
```
type:    <class 'list'>
devices: [device(type='cpu'), device(type='cpu')]
```

**After fix** (nn.ModuleList in `_base_model_v2.py`):
```
type:    <class 'torch.nn.modules.container.ModuleList'>
devices: [device(type='cuda', index=0), device(type='cuda', index=0)]
```

**Comment by StrikerEureka34** [comment]
@Gitanaskhan26 why did you raise a PR? We were discussing and ascertaining the scope of the issue here.
Even that aside you should ask the author first if he is working on the issue or not before opening a PR.

**Comment by StrikerEureka34** [comment]
So I spent a lot of time testing and figuring out how to actually test this without a GPU in CI. Turns out `state_dict()` doesn't work here because torchmetrics registers its state with persistent=False by default, so nothing shows up there. 

But what ended up working was torch.device("meta"), it's a real device move so if '.to()' doesn't propagate to the metrics, their tensors stay on cpu and the assertion catches it. 
Seemed like a cool and clean way to replicate the GPU crash on CPU-only CI, I think?

I've raised a draft pr for the same, looking forward for your thoughts, thanks! @fnhirwa @phoeenniixx

**Comment by fnhirwa** [comment]
Thanks @StrikerEureka34, I'm able to reproduce the bug, will look at your PR and see if I can add some pointers

**Comment by StrikerEureka34** [comment]
> Thanks [@StrikerEureka34](https://github.com/StrikerEureka34), I'm able to reproduce the bug, will look at your PR and see if I can add some pointers

Much appreciated, thanks!
